master/getting-started/kubernetes/installation/migration-from-flannel.md

```diff
@@ -0,0 +1,127 @@
+- Your cluster must allow for modification and removal of the flannel daemon set. For example, it must not be installed using the Kubernetes add-on manager.
+
+# Procedure
+
+1. First, install Calico.
+
+   ```
+   kubectl apply -f {{site.url}}/{{page.version}}/manifests/flannel-migration/calico.yaml
+   ```
+
+   Then, install the migration controller to initiate the migration.
+
+   ```
+   kubectl apply -f {{site.url}}/{{page.version}}/manifests/flannel-migration/migration-job.yaml
+   ```
+
+   Once applied, you will see the nodes begin to update one at a time.
+
+1. To monitor the migration, run the following command.
+
+   ```
+   kubectl get jobs -n kube-system flannel-migration
+   ```
+
+   Note that the migration controller may be rescheduled one or more times during the migration as the node hosting
+   the controller is upgraded. The installation will be complete when the ouput of the above command shows `1/1` completions. For example:
+
+   ```
+   NAME                COMPLETIONS   DURATION   AGE
+   flannel-migration   1/1           2m59s      5m9s
+   ```
+
+1. Once complete, remove the migration controller with the following command.
+
+   ```
+   kubectl delete -f {{site.url}}/{{page.version}}/manifests/flannel-migration/migration-job.yaml
+   ```
+
+# Troubleshooting
+
+## Checking migration status
+
+The migration controller should run to completion. You can run the following command to view the controller's current status.
+
+```
+kubectl get pods -n kube-system -l k8s-app=flannel-migration-controller
+```
+
+## Viewing migration logs
+
+The migration controller logs can be viewed with the following command:
+
+```
+kubectl logs -n kube-system -l k8s-app=flannel-migration-controller
+```
+
+The logs will indicate if any actions need to be taken by the user.
+
+# Rollback
+
+Migration from Calico to flannel is not supported. If a problem is encountered during the migration, you can use the following steps
+to rollback the migration.
+
+1. Remove the migration controller and Calico.
+
+   ```
+   kubectl delete -f {{site.url}}/{{page.version}}/manifests/flannel-migration/migration-job.yaml
+   kubectl delete -f {{site.url}}/{{page.version}}/manifests/flannel-migration/calico.yaml
+   ```
+
+1. Determine the nodes which have been migrated to Calico.
+
+   ```
+   kubectl get nodes -l projectcalico.org/node-network-during-migration=calico
+   ```
+
+Then, for each node found above, run the following commands to remove Calico.
+
+1. Cordon and drain the node.
+
+   ```
+   kubectl drain node <node name>
+   ```
+
+1. Log into the machine and remove the CNI configuration.
+
+   ```
+   rm /etc/cni/net.d/10-calico.conflist
+   ```
+
+1. Reboot the node.
+
+1. Enable flannel on the node.
+
+   ```
+   kubectl label node <node name> projectcalico.org/node-network-during-migration=flannel
+   ```
+
+1. Uncordon the node.
+
+   ```
+   kubectl uncordon node <node name>
+   ```
```

Comment:
At the end, user still need to perform two steps in order to deploy Flannel to new node.
- Remove nodeSelector "projectcalico.org/node-network-during-migration=flannel" from Flannel Daemonset. This may cause Flannel daemonset pod restarts.
- Remove node label "projectcalico.org/node-network-during-migration" from all nodes.